plugins/modules/filesystem.py

```diff
@@ -197,6 +198,121 @@ def fs_state(module, filesystem):
     return False
 
 
+def check_attr_change(module, filesystem):
+    """
+    Determines if changes will be made on the filesystem.
+    param module: Ansible module argument spec.
+    param filesystem: filesystem name.
+    return: True - changes will be made on the filesystem / False = filesystem will remain unchanged
+    """
+
+    cmd = "lsfs -cq %s" % filesystem
+    rc, stdout, stderr = module.run_command(cmd)
+    all_attr = stdout.splitlines()
+    old_attr = all_attr[1].split(":")
+
+    # check for extended attributes
+    old_ext_attr = dict()
+    if len(all_attr) == 3:
+        attrs = re.sub(r"[()]", "", all_attr[2]).strip()
+        attrs = attrs.split(":")
+        for attr in attrs:
+            attr = attr.rsplit(" ", 1)
+            key = re.sub(" ", "_", attr[0]).lower()
+            old_ext_attr[key] = attr[1]
+
+    new_attr = dict()
+    attrs = module.params["attributes"]
+    if attrs:
+        for attr in attrs:
+            attr = attr.strip()
+            attr = attr.split("=")
+            new_attr[attr[0]] = attr[1]
+
+    # check if mount group changed
+    old_mnt_grp = old_attr[4]
+    new_mnt_grp = module.params["mount_group"]
+    if new_mnt_grp:
+        if new_mnt_grp != old_mnt_grp:
+            return True
+
+    # check if permissions changed
+    old_perms = old_attr[6].split(",")[0]
```

Comment:
Maybe you should retrieve the old values only if the new value exists

Reply:
the old values are retrieved one time at the beginning of the function.

Reply:
`# lsfs -cq /tmp/issue76

#MountPoint:Device:Vfs:Nodename:Type:Size:Options:AutoMount:Acct
/tmp/issue76:/dev/fslv01:jfs2::test_mount:10485760:rw:yes:no
  (lv size 10485760:fs size 10485760:block size 4096:sparse files yes:inline log no:inline log size 0:EAformat v1:Quota no:DMAPI no:VIX yes:EFS no:ISNAPSHOT no:MAXEXT 0:MountGuard no:LFF no)`

Reply:
here is a sample output of lsfs -cq <filesystem>, this is the output of a jfs2 fs

Reply:
retrieve the old_perms as it will only be used if the new_perms have values in line 242. Kind of save the cyles to only if new_perms are there.

Reply:
i am not quite understanding. the info for old permission is retrieved at the beginning with lsfs -cq together with the other attributes and extended attributes.

are you talking about retrieving old_perms from old_attr list?
or are you talking about retrieving the info for old permissions from the remote module?

Reply:
I meant that you read the old_mnt_grp = old_attr[4], then check if the new_mnt_grp returns a value. If it does, then you compare the new_mt_grp with the old_mnt_grp. I was saying that if new_mnt_grp does not exist, you don't need the old_mnt_grp at all. Why not retrieve the old_mnt_grp only if the new_mnt_grp ixist, then compare it.

Reply:
right, so basically move that retrieval inside the if statement i have for checking if new_mnt_grp exists. yes i can certainly do that. i think that can be applied to the other checks as well in a similar fashion.